src/panels/lovelace/cards/hui-entity-button-card.js

```diff
@@ -0,0 +1,162 @@
+import { html } from "@polymer/polymer/lib/utils/html-tag.js";
+import { PolymerElement } from "@polymer/polymer/polymer-element.js";
+import { fireEvent } from "../../../common/dom/fire_event.js";
+
+import "../../../components/ha-card.js";
+
+import toggleEntity from "../common/entity/toggle-entity.js";
+import stateIcon from "../../../common/entity/state_icon.js";
+import computeStateDomain from "../../../common/entity/compute_state_domain.js";
+import computeStateName from "../../../common/entity/compute_state_name.js";
+import EventsMixin from "../../../mixins/events-mixin.js";
+import LocalizeMixin from "../../../mixins/localize-mixin.js";
+
+/*
+ * @appliesMixin EventsMixin
+ */
+class HuiEntityButtonCard extends LocalizeMixin(EventsMixin(PolymerElement)) {
+  static get template() {
+    return html`
+    <style>
+      ha-icon {
+        display: flex;
+        margin: auto;
+        width: 40%;
+        height: 40%;
+        color: var(--paper-item-icon-color, #44739e);
+      }
+      ha-icon[data-domain=light][data-state=on],
+      ha-icon[data-domain=switch][data-state=on],
+      ha-icon[data-domain=binary_sensor][data-state=on],
+      ha-icon[data-domain=fan][data-state=on],
+      ha-icon[data-domain=sun][data-state=above_horizon] {
+        color: var(--paper-item-icon-active-color, #FDD835);
+      }
+      ha-icon[data-state=unavailable] {
+        color: var(--state-icon-unavailable-color);
+      }
+      state-badge {
+        display: flex;
+        margin: auto;
+        width:40%;
+        height:40%;
+      }
+      paper-button {
+        display: flex;
+        margin: auto;
+        text-align: center;
+      }
+      .not-found {
+        flex: 1;
+        background-color: yellow;
+        padding: 8px;
+      }
+    </style>
+    <ha-card on-click="_handleClick">
+      <paper-button>
+        <div class$="[[_computeClassName(_stateObj)]]">
+          <ha-icon id="ButtonIcon"
+            data-domain$="[[_computeDomain(_stateObj)]]"
+            data-state$="[[_stateObj.state]]"
+            icon="[[_computeIcon(_stateObj)]]"></ha-icon>
+          <span>[[_computeName(_stateObj)]]</span>
+        </div>
+      </paper-button>
+    </ha-card>
+    `;
+  }
+
+  static get properties() {
+    return {
+      hass: {
+        type: Object,
+      },
+      _config: Object,
+      _stateObj: {
+        type: Object,
+        computed: "_computeStateObj(hass.states, _config.entity)",
+        observer: "stateObjChanged",
+      },
+    };
+  }
+
+  getCardSize() {
+    return 2;
+  }
+
+  setConfig(config) {
+    if (!config || !config.entity) {
+      throw new Error("Invalid card configuration");
+    }
+    const defaults = { show_state: true };
+    this._config = { ...defaults, ...config };
```

Comment:
So you should either move `defaults` to a constant above the ckass definition or you move it inside the splat:

```js
this._config = {
  show_state: true,
  ...config
}
```

the latter has the preference because it means that code related to defaults is only run if we instantiate the class, not when we merely load the file.